vault/vault.go

```diff
@@ -79,7 +92,23 @@ func New(
 	if err != nil {
 		return nil, err
 	}
-	client.SetToken(token)
+
+	var autoAuth bool
+	if getVaultParam(secretConfig, AuthMethod) == AuthMethodKubernetes {
+		token, err := kubernetesAuth(client, secretConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		autoAuth = true
+		client.SetToken(token)
+	} else {
+		token := getVaultParam(secretConfig, api.EnvVaultToken)
+		if token == "" {
+			return nil, ErrVaultTokenNotSet
+		}
+		client.SetToken(token)
```

Comment:
client.SetToken can be moved out of if and else.

Reply:
Updated.